src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -99,20 +101,33 @@ object Task {
     */
   case class ReservationWithVolumes(volumeIds: Iterable[LocalVolumeId])
 
-  case class LocalVolumeId(idString: String) {
+  case class LocalVolume(id: LocalVolumeId, persistentVolume: PersistentVolume)
+
+  case class LocalVolumeId(appId: PathId, containerPath: String, uuid: String) {
+    import LocalVolumeId._
+    lazy val idString = appId.safePath + appDelimiter + containerPath + volumeDelimiter + uuid
+
     override def toString: String = s"LocalVolume [$idString]"
```

Comment:
Suggestion: `def toString = idString`
The compiler does not complain about string coercion.